### PR TITLE
Prometheus metrics

### PR DESCRIPTION
**Pull request by gabzim**
Missing from this PR: Strategy so plugins can add their own metrics avoiding naming collisions. I believe if they create prometheus metrics they should be exposed automatically but two plugins could collide if they pick the same name.

### CODE REVIEW

File: relayer-engine/src/executor/executorHarness.ts [hunk 1]
```diff
@@ -68,9 +69,53 @@ export async function run(plugins: Plugin[], storage: Storage) {
...
     }),
   );
-  await spawnExecutor(storage, plugins, providers, workerInfoMap, logger);
+
+  maxActiveWorkflowsGauge.set(MAX_ACTIVE_WORKFLOWS);
+  spawnExecutor(storage, plugins, providers, workerInfoMap, logger);
+  logger.debug("End of executor harness run function");
 }
 
+const inProgressWorkflowsGauge = new Gauge({
```
Comment (JoeHowarth) [inline relayer-engine/src/executor/executorHarness.ts:78]: nit: what do you think about moving theses metrics to `executor/metrics.ts`?

File: relayer-engine/src/listener/eventHarness.ts [hunk 2]
```diff
@@ -12,6 +13,24 @@ const logger = () => {
   return _logger;
 };
 
+const receivedEventsCounter = new Counter({
```
Comment (JoeHowarth) [inline relayer-engine/src/listener/eventHarness.ts:16]: nit: same as for executor, how about moving these metric definitions to `listener/metrics.ts`?

File: example-project/plugins/dummy_plugin/src/plugin.ts [hunk 3]
```diff
@@ -36,6 +37,9 @@ interface WorkflwoPayloadDeserialized {
   count: number;
 }
 
+const randomInt = (min: number, max: number) =>
```
Comment (JoeHowarth) [inline example-project/plugins/dummy_plugin/src/plugin.ts:40]: Inspired https://github.com/wormhole-foundation/relayer-engine/issues/30#issue-1521214247

Comment (gabzim) [reply example-project/plugins/dummy_plugin/src/plugin.ts:40]: Agreed.

Comment (gabzim) [reply example-project/plugins/dummy_plugin/src/plugin.ts:40]: should I do that in this PR or do you want to tackle separatedly?

Comment (JoeHowarth) [reply example-project/plugins/dummy_plugin/src/plugin.ts:40]: One of us should do it separately 👍

**Pull request merged**